### PR TITLE
doc(README.md): add documentation on configuring the controller for object storage

### LINKED ISSUE #97: Add documentation for configuring the controller's object storage
After https://github.com/deis/controller/pull/523, the controller will read the `minio-user` secret from the k8s API, but it won't read any object storage endpoint data. It simply passes the secret on to [slugrunner](https://github.com/deis/slugrunner) so that the slugrunner can download the slug (note that the builder passes the slug URL through workflow directly to the slugrunner).

Instructions for how to configure this secret for use in the controller/slugrunner should be added to https://github.com/deis/workflow/blob/master/src/installing-workflow/configuring-object-storage.md#deisdatabase

### PR DESCRIPTION
**Pull request by arschles**
Fixes #97 
Requires https://github.com/deis/controller/pull/523

Ref https://github.com/deis/deis/issues/4962

### CODE REVIEW

File: src/installing-workflow/configuring-object-storage.md [hunk 1]
```diff
@@ -89,6 +90,20 @@ The slugrunner reads credentials from the below locations on the filesystem.
...
+
+When the controller needs to launch a new buildpack application or scale one up, it uses a [replication controller](http://kubernetes.io/docs/user-guide/replication-controller/). Since the slugrunner needs to download the slug to run, it needs the object storage location of the slug and the object storage credentials.
+
+### Environment Variables
+
+The controller needs no environment variables for object storage configuration.
+
+### Credentials
+
+Since the object storage location information comes from the builder, the controller only needs access to the credentials information. The controller gets this information by accessing the `minio-user` secret directly from the Kubernetes API.
```
Comment (krancour) [inline src/installing-workflow/configuring-object-storage.md]: > The controller gets this information by accessing the `minio-user` secret

Even in cases where you're not actually using minio?  i.e. This is the secret it uses, regardless?

Comment (arschles) [inline src/installing-workflow/configuring-object-storage.md]: Yes

File: src/installing-workflow/configuring-object-storage.md [hunk 2]
```diff
@@ -89,6 +90,20 @@ The slugrunner reads credentials from the below locations on the filesystem.
 - Key: `/var/run/secrets/object/store/access-key-id`
 - Secret: `/var/run/secrets/object/store/access-key-secret`
 
+## [deis/controller](https://github.com/deis/controller)
+
+When the controller needs to launch a new buildpack application or scale one up, it uses a [replication controller](http://kubernetes.io/docs/user-guide/replication-controller/). Since the slugrunner needs to download the slug to run, it needs the object storage location of the slug and the object storage credentials.
```
Comment (carmstrong) [inline src/installing-workflow/configuring-object-storage.md]: "needs to launch or scale a new buildpack application"? `scale one up` reads awkwardly.

File: src/installing-workflow/configuring-object-storage.md (hunk 1 above)
Comment (arschles) [inline src/installing-workflow/configuring-object-storage.md]: Noted in 30c52484aae1dc0ae1d409fca3ec02b33930fa18

Comment (krancour) [inline src/installing-workflow/configuring-object-storage.md]: I'm starting to understand where some of my confusion arose from yesterday.  I think I had viewed the steps for _not_ using minio as starting with step 1: rip out everything minio-related... in reality, I guess step 1 (currently) is: pretend you're using minio and put your s3 (or whatever) creds where the minio creds usually go.  This is enlightening.

File: src/installing-workflow/configuring-object-storage.md (hunk 2 above)
Comment (arschles) [inline src/installing-workflow/configuring-object-storage.md]: Good catch. Reworded in 45ef1df464911087a12760b2217135d1910b748c

File: src/installing-workflow/configuring-object-storage.md (hunk 1 above)
Comment (arschles) [inline src/installing-workflow/configuring-object-storage.md]: Glad to hear it.

As a sidenote - the naming & configuration steps will get better & significantly easier in the next release. https://github.com/deis/deis/issues/4966 has details on the plan.

**Pull request merged**